src/main/java/org/generationcp/middleware/api/germplasm/GermplasmAttributeServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package org.generationcp.middleware.api.germplasm;
+
+import org.generationcp.middleware.api.attribute.AttributeDTO;
+import org.generationcp.middleware.domain.germplasm.GermplasmAttributeDto;
+import org.generationcp.middleware.domain.germplasm.GermplasmAttributeRequestDto;
+import org.generationcp.middleware.hibernate.HibernateSessionProvider;
+import org.generationcp.middleware.manager.DaoFactory;
+import org.generationcp.middleware.pojos.Attribute;
+import org.generationcp.middleware.pojos.UDTableType;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class GermplasmAttributeServiceImpl implements GermplasmAttributeService {
+
+	private final DaoFactory daoFactory;
+
+	public GermplasmAttributeServiceImpl(final HibernateSessionProvider sessionProvider) {
+		this.daoFactory = new DaoFactory(sessionProvider);
+	}
+
+	@Override
+	public List<GermplasmAttributeDto> getGermplasmAttributeDtos(final Integer gid, final String attributeType) {
+		return this.daoFactory.getAttributeDAO().getGermplasmAttributeDtos(gid, attributeType);
+	}
+
+	@Override
+	public Integer createGermplasmAttribute(final Integer gid, final GermplasmAttributeRequestDto dto, final Integer userId) {
+		final Attribute newAttribute = new Attribute();
+		newAttribute.setGermplasmId(gid);
+		newAttribute.setTypeId(this.daoFactory.getUserDefinedFieldDAO().getByTableTypeAndCode(UDTableType.ATRIBUTS_ATTRIBUTE.getTable(),
+			dto.getAttributeType(), dto.getAttributeCode()).getFldno());
+		newAttribute.setUserId(userId);
+		newAttribute.setAval(dto.getValue());
+		newAttribute.setLocationId(dto.getLocationId());
+		newAttribute.setReferenceId(0);
+		newAttribute.setAdate(Integer.valueOf(dto.getDate()));
+		return this.daoFactory.getAttributeDAO().save(newAttribute).getAid();
+	}
+
+	@Override
+	public void updateGermplasmAttribute(final Integer attributeId, final GermplasmAttributeRequestDto dto) {
+		final Attribute attribute = this.daoFactory.getAttributeDAO().getById(attributeId);
+		attribute.setAval(dto.getValue());
+		attribute.setLocationId(dto.getLocationId());
+		attribute.setAdate(Integer.valueOf(dto.getDate()));
```

Comment:
@mcrimi - do we update the `atributs.adate` when updating the attribute?

Reply:
According to the mockup it is actually possible to modify the date, It seems it is not a system date but a date set by breeders